src/components/preferences/Preferences.js

```diff
@@ -383,8 +384,14 @@ function Preferences(props) {
             </>
         );
     }
-    const validateShortCuts = (values, props) => {
+    const validate = (values, props) => {
         const errors = {};
+        if (values?.webhook?.url) {
+            const type = values?.webhook?.type?.type;
+            if (type === null || type === undefined || type === '') {
+                errors["webhook.type.type"] =  t("webhookTypeError");
+            }
+        }
```

Comment:
This validation will run but will not mark errors on the form 😭

Reply:
Going off memory, I think `errors` is an object with the same format as the model you provided. 
So `errors["webhook.type.type"]` would be accessing an object like `{"webhook.type.type": "someValue"}` instead of `{webhook: {type: {type: "someValue"}}}`.  So I think if you update this to `errors.webhook.type.type` you'll get the result you want.

Reply:
Ahh that makes sense although `errors.webhook.type.type = err msg` resulted in error. So I guess I have to construct the error object manually 🤔

Reply:
Actually that reminds me! I don't think it's advertised anywhere, but Formik has a `setIn` function (similar to the `getIn` function that we're using in our helper form fields) that would be useful in this case.

```
errors = setIn(errors, "webhook.type.type", t("webhookTypeError"))
```

Reply:
`errors = { webhook: { type: { type: t("webhookTypeError") } } };` worked for me 🎉